charts/stable/docker-compose/values.yaml

```
@@ -60,7 +41,24 @@ workload:
           env:
             DOCKER_TLS_CERTDIR: /certs
             COMPOSE_FILE: ""
-
+          lifecycle:
+            postStart:
+              type: exec
+              command:
+                - /bin/bash
+                - -c
+                - |
+                  if [ "${COMPOSE_FILE}" ]; then
+                    counter=0;
+                    while ! docker info > /dev/null 2>&1 && [[ ${counter} -lt 10 ]]; do
+                      echo "docker not running yet. Waiting..." >> /proc/1/fd/1;
+                      sleep 5;
+                      counter=$((counter+1));
+                    done
+                    docker-compose -f "${COMPOSE_FILE}" up -d && echo "Loading COMPOSE_FILE at location ${COMPOSE_FILE}" >> /proc/1/fd/1;
+                  else
+                    echo "COMPOSE_FILE not set, not auto-loading compose-file" >> /proc/1/fd/1;
+                  fi
     type: StatefulSet
     replicas: 1
     strategy: RollingUpdate
```

```suggestion
    strategy: Recreate
```
I don't think docker-compose will appreciate rolling updates

---

`Recreate` sounds more appropriate, but I don't know enough about the implications of changing this.